plugin/storage/grpc/config/config.go

```diff
@@ -37,10 +42,34 @@ type Configuration struct {
 	RemoteTLS            tlscfg.Options
 	RemoteConnectTimeout time.Duration `yaml:"connection-timeout" mapstructure:"connection-timeout"`
 	TenancyOpts          tenancy.Options
+}
+
+type ConfigV2 struct {
+	Configuration
+	TenancyOpts                    tenancy.Options
+	configgrpc.ClientConfig        `mapstructure:",squash"`
+	exporterhelper.TimeoutSettings `mapstructure:",squash"`
 
 	remoteConn *grpc.ClientConn
```

Comment:
can we avoid having runtime objects in the configuration structs? It was a bad pattern in v1

Reply:
moved it in `factory` struct